zephyr/lib/regions_mm.c

```diff
@@ -216,6 +216,124 @@ struct vmh_heap *vmh_init_heap(const struct vmh_heap_config *cfg,
 	return NULL;
 }
 
+/**
+ * @brief Checks if region of a sys_mem_blocks have any allocated block
+ *
+ * @param blocks Pointer to the sys_mem_blocks allocator
+ * @param ptr Pointer to the memory region to be checked
+ * @param size Size of memory region
+ *
+ * @retval true if there is any allocation in the queried region
+ */
+static inline bool vmh_is_region_used(struct sys_mem_blocks *blocks, const uintptr_t ptr,
+				      const size_t size)
+{
+	__ASSERT_NO_MSG(IS_ALIGNED(size, 1 << blocks->info.blk_sz_shift));
+	return !sys_mem_blocks_is_region_free(blocks, UINT_TO_POINTER(ptr),
+					      size >> blocks->info.blk_sz_shift);
+}
+
+/**
+ * @brief Checks whether the memory region is located on memory pages unused by other allocations
+ *
+ * @param allocator Pointer to the sys_mem_blocks allocator
+ * @param ptr Pointer to the memory region to be checked
+ * @param size Size of memory region
+ * @param begin Address to a variable where address of the first unused memory page will be placed
+ * @param out_size Address to a variable where size of unused memory pages will be placed
+ *
+ * @retval true if there is any unused memory page
+ */
+static bool vmh_get_map_region_boundaries(struct sys_mem_blocks *blocks, const void *ptr,
+					  const size_t size, uintptr_t *begin, size_t *out_size)
+{
+	__ASSERT_NO_MSG(blocks);
+	__ASSERT_NO_MSG(begin);
+	__ASSERT_NO_MSG(out_size);
+
+	const size_t block_size = 1 << blocks->info.blk_sz_shift;
+	const size_t size_aligned = ALIGN_UP(size, block_size);
+	uintptr_t addr = ALIGN_DOWN((uintptr_t)ptr, CONFIG_MM_DRV_PAGE_SIZE);
+	uintptr_t addr_end = ALIGN_UP((uintptr_t)ptr + size, CONFIG_MM_DRV_PAGE_SIZE);
+	const uintptr_t size_before = (uintptr_t)ptr - addr;
+	const uintptr_t size_after = addr_end - (uintptr_t)ptr - size_aligned;
+
+	__ASSERT_NO_MSG(IS_ALIGNED(size_before, block_size));
+
+	if (size_before && vmh_is_region_used(blocks, addr, size_before))
+		/* skip first page */
+		addr += CONFIG_MM_DRV_PAGE_SIZE;
+
+	if (size_after && vmh_is_region_used(blocks, POINTER_TO_UINT(ptr) + size_aligned,
+					     size_after))
+		/* skip last page */
+		addr_end -= CONFIG_MM_DRV_PAGE_SIZE;
+
+	if (addr < addr_end) {
+		*begin = addr;
+		*out_size = addr_end - addr;
+		return true;
+	}
```

Comment:
```
if (addr >= addr_end)
	return false;
```

Reply:
addr_end is always bigger from addr by size , condition is unnecessary